### PR TITLE
feat: specify components' platform

### PR DESCRIPTION
**Pull request by LG95**
### Problem
Some components aren't compatible with all platforms. 
Sometimes, a component is only desired to be used in some platforms.

### Solution
Add `BeaglePlatformWrapper` class in the backend to wrap components adding platform specification. The backend uses it to match against the new `beagle-platform` header which each platform sends to identify itself, removing all unmatched components then unwrapping the matched components before sending the response.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by uziasferreirazup** [comment]
what do you think about changing this approach? I think that this type of logic does not need to have in the Android beagle and the iOS beagle, we could isolate this only in the bff and it send the right json to the respective platforms and this attribute does not exist on the platforms themselves

**Comment by LG95** [comment]
@uziasferreirazup how's this?

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/mapper/ScreenRequestMapper.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ internal object ScreenRequestMapper {
         return RequestData(
             uri = URI(newUrl),
             method = method,
-            headers = screenRequest.headers,
+            originalHeaders = screenRequest.headers,
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/mapper/ScreenRequestMapper.kt:44]: Looking for what this attribute does, I think its not necessary to create extra attribute and use the existing headers.

Comment (LG95) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/mapper/ScreenRequestMapper.kt:44]: This features needs to always add the beagle-platform header, how do you suggest I do this without this attribute?

**Pull request merged**